tests/test_server.py

```diff
@@ -54,6 +55,9 @@ def setUpClass(cls):
         cls.server_thread = threading.Thread(target=cls.httpd.serve_forever)
         cls.server_thread.daemon = True
         cls.server_thread.start()
+        # Prevent user settings from being loaded and interfere with test runs
+        cls.get_settings_patcher = patch("mlx_lm.server.get_settings", return_value={})
+        cls.mock_get_settings = cls.get_settings_patcher.start()
```

Comment:
I'm not sure if this is good approach. It defo highlights an issue introduced by a global settings. Maybe a more proper mocking should be used instead, but honestly, I could not get it working in a more global way. (Python is not my native language.) My concern with this is that this patterns should become repeated in all tests for any modules which use the settings file.